src/prefect/tasks/monte_carlo/client.py

```diff
@@ -0,0 +1,205 @@
+import pendulum
+import requests
+from typing import Any, Dict, List
+
+from prefect.utilities.logging import get_logger
+
+
+class MonteCarloClient:
+    def __init__(
+        self,
+        api_key_id: str,
+        api_token: str,
+    ) -> None:
+        self.api_key_id = api_key_id
+        self.api_token = api_token
+        self.logger = get_logger()
+        self._api_url = "https://api.getmontecarlo.com/graphql"
+
+    def _send_graphql_request(
+        self, query: str, variables: dict = None
+    ) -> Dict[str, Any]:
+        response = requests.post(
+            url=self._api_url,
+            json=dict(query=query, variables=variables),
+            headers={
+                "x-mcd-id": self.api_key_id,
+                "x-mcd-token": self.api_token,
+                "Content-Type": "application/json",
+            },
+        )
+        response.raise_for_status()
```

Comment:
You could use `format_graphql_request_error` as used in the Prefect `Client` to log something helpful here

Example:

```python
        # Check if request returned a successful status
        try:
            response.raise_for_status()
        except requests.HTTPError as exc:
            if response.status_code == 400 and params and "query" in params:
                # Create a custom-formatted err message for graphql errors which always
                # return a 400 status code and have "query" in the parameter dict
                try:
                    graphql_msg = format_graphql_request_error(response)
                except Exception:
                    # Fallback to a general message
                    graphql_msg = (
                        "This is likely caused by a poorly formatted GraphQL query or "
                        "mutation but the response could not be parsed for more details"
                    )
                raise ClientError(f"{exc}\n{graphql_msg}") from exc

            # Server-side and non-graphql errors will be raised without modification
            raise
```

Reply:
(this isn't strictly necessary, you're welcome not to do this)